tests/test_dirty_local.py

```diff
@@ -0,0 +1,117 @@
+import os
+from pathlib import Path
+from shutil import copy2, copytree
+
+import pytest
+from plumbum import local
+from plumbum.cmd import git
+
+import copier
+from copier.errors import DirtyLocalWarning
+from copier.main import run_copy, run_update
+
+from .helpers import DATA, PROJECT_TEMPLATE, build_file_tree, filecmp
+
+
+def test_copy(tmp_path_factory):
+    src, dst = map(tmp_path_factory.mktemp, ("src", "dst"))
+
+    # dirs_exist_ok not available in Python 3.7
+    for item in os.listdir(PROJECT_TEMPLATE):
+        s = os.path.join(PROJECT_TEMPLATE, item)
+        d = os.path.join(src, item)
+        if os.path.isdir(s):
+            copytree(s, d)
+        else:
+            copy2(s, d)
+
+    with local.cwd(src):
+        git("init")
+
+    with pytest.warns(DirtyLocalWarning):
+        copier.copy(str(src), str(dst), data=DATA, quiet=True)
+
+    generated = (dst / "pyproject.toml").read_text()
+    control = (Path(__file__).parent / "reference_files/pyproject.toml").read_text()
+    assert generated == control
+
+    # assert template still dirty
+    with local.cwd(src):
+        assert bool(git("status", "--porcelain").strip())
+
+
+# Will fail due to lingering deleted file.
+# HACK https://github.com/copier-org/copier/issues/461
+# TODO Remove xfail decorator when fixed.
+@pytest.mark.xfail
```

Comment:
```suggestion
@pytest.mark.xfail(strict=True)
```

This way we won't forget to do what the comment says.